dll/win32/shell32/shlexec.cpp

```diff
@@ -1824,6 +1824,14 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         if(l > dwApplicationNameLen) dwApplicationNameLen = l + 1;
         wszApplicationName = (LPWSTR)HeapAlloc(GetProcessHeap(), 0, dwApplicationNameLen * sizeof(WCHAR));
         memcpy(wszApplicationName, sei_tmp.lpFile, l * sizeof(WCHAR));
+
+        if (wszApplicationName[2] == 0 && wszApplicationName[1] == L':' &&
+            ((L'A' <= wszApplicationName[0] && wszApplicationName[0] <= L'Z') ||
+             (L'a' <= wszApplicationName[0] && wszApplicationName[0] <= L'z')))
```

Comment:
why not use ```isalpha``` function? (to be explicit) //defined in ctype.h (for C) or cctype (for C++)

Reply:
isalpha is locale-dependent

Reply:
Wow, I don't know this before. Thanks!